sdk/communication/communication-short-codes/src/shortCodesClient.ts

```diff
@@ -0,0 +1,139 @@
+import { ShortCodes as GeneratedClient } from "./generated/src/operations";
+import {
+  ShortCode,
+  ShortCodesGetShortCodesOptionalParams,
+  ShortCodesUpsertUSProgramBriefOptionalParams,
+  USProgramBrief
+} from "./generated/src/models/";
+
+/**
+ * Client options used to configure the ShortCodesClient API requests.
+ */
+export interface ShortCodesClientOptions extends PipelineOptions {}
+
+const isShortCodesClientOptions = (options: any): options is ShortCodesClientOptions =>
+  options && !isKeyCredential(options) && !isTokenCredential(options);
+
+export class ShortCodesClient {
+  /**
+   * A reference to the auto-generated ShortCodes HTTP client.
+   */
+  private readonly client: GeneratedClient;
+
+  public constructor(connectionString: string, options?: ShortCodesClientOptions);
+
+  public constructor(url: string, credential: KeyCredential, options?: ShortCodesClientOptions);
+
+  public constructor(url: string, credential: TokenCredential, options?: ShortCodesClientOptions);
+
+  public constructor(
+    connectionStringOrUrl: string,
+    credentialOrOptions?: KeyCredential | TokenCredential | ShortCodesClientOptions,
+    maybeOptions: ShortCodesClientOptions = {}
+  ) {
+    const { url, credential } = parseClientArguments(connectionStringOrUrl, credentialOrOptions);
+    const options = isShortCodesClientOptions(credentialOrOptions)
+      ? credentialOrOptions
+      : maybeOptions;
+    const libInfo = `azsdk-js-communication-short-codes/${SDK_VERSION}`;
+
+    if (!options.userAgentOptions) {
+      options.userAgentOptions = {};
+    }
+
+    if (options.userAgentOptions.userAgentPrefix) {
+      options.userAgentOptions.userAgentPrefix = `${options.userAgentOptions.userAgentPrefix} ${libInfo}`;
+    } else {
+      options.userAgentOptions.userAgentPrefix = libInfo;
+    }
+
+    const internalPipelineOptions: InternalPipelineOptions = {
+      ...options,
+      ...{
+        loggingOptions: {
+          logger: logger.info
+        }
+      }
+    };
+
+    const authPolicy = createCommunicationAuthPolicy(credential);
+    const pipeline = createPipelineFromOptions(internalPipelineOptions, authPolicy);
+    this.client = new ShortCodesGeneratedClient(url, pipeline).shortCodes;
+  }
+
+  public listShortCodes(
+    options: ShortCodesGetShortCodesOptionalParams = {}
+  ): PagedAsyncIterableIterator<ShortCode> {
+    const { span, updatedOptions } = createSpan("ShortCodesClient-listShortCodes", options);
+    const iter = this.client.listShortCodes(updatedOptions);
+    span.end();
+    return iter;
+  }
+
+  public async upsertUSProgramBrief(
+    programBriefId: string,
+    options: ShortCodesUpsertUSProgramBriefOptionalParams = {}
+  ): Promise<RestResponse> {
+    const { span, updatedOptions } = createSpan("ShortCodesClient-upsertUSProgramBrief", options);
+    const promise = await this.client.upsertUSProgramBrief(programBriefId, updatedOptions);
```

Comment:
IIRC convention is to wrap calls to the generated client in try..catch..finally like [this](https://github.com/Azure/azure-sdk-for-js/blob/74b93c7fb8078de1f309a00447ee617d9a1764ea/sdk/communication/communication-phone-numbers/src/phoneNumbersClient.ts#L200). Same for other methods.